include/clap/ext/draft/context-menu.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include "../../plugin.h"
+
+// This extension lets the host and plugin exchange menu items and let the plugin ask the host to
+// show its context menu.
+
+static CLAP_CONSTEXPR const char CLAP_EXT_CONTEXT_MENU[] = "clap.context-menu.draft/0";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// There can be different target kind for a context menu
+enum {
+   CLAP_CONTEXT_MENU_TARGET_KIND_GLOBAL = 0,
+   CLAP_CONTEXT_MENU_TARGET_KIND_PARAM = 1,
+};
+
+// Describes the context menu target
+typedef struct clap_context_menu_target
+{
+   uint32_t kind;
+   clap_id id;
+} clap_context_menu_target_t;
+
+// Flags for the context menu accelerator
+enum {
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_SHIFT = (1 << 0),
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_CONTROL = (1 << 1),
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_ALT = (1 << 2),
+
+   // Windows key or Command key
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_META1 = (1 << 3),
+};
+
+// Describes a single key strike with its modifiers
+typedef struct clap_context_menu_accelerator_key {
+   uint32_t flags;
+   uint32_t key; // ASCII key
+} clap_context_menu_accelerator_key_t;
+
+// Describes an accelerator which can consist of a key strike combo
+typedef struct clap_context_menu_accelerator {
+   uint32_t combo_size;
+   clap_context_menu_accelerator_key_t keys[4];
+} clap_context_menu_accelerator_t;
```

Comment:
Yes this is just a display thing and nothing else.
Because the plugin will handle its windowing events itself, same for the host.

The benefit is that the way those accelerators are shown in the menu is consistent.

I'm OK to remove the accelerators, I think it clutters the interface.

Reply:
Yeah if it's just purely for display purposes I'd also be fine with removing it. We should have a proper vote for that first of course. Realistically the odds of, say, Surge XT's context menu showing relevant global Bitwig shortcut keys (especially since most context menus and context menu items will be specific to a parameter) are very slim. Or the other way around.